src/event.c

```diff
@@ -100,7 +100,10 @@ static void x_check_cb(EV_P_ ev_check *w, int revents) {
             last_cursor_pos.y = root_y;
         }
 
-        if (config.hide_on_touch && (cookie->evtype == XI_RawTouchBegin || cookie->evtype == XI_RawTouchUpdate)) {
+        if (
+               (config.hide_on_touch && (cookie->evtype == XI_RawTouchBegin || cookie->evtype == XI_RawTouchUpdate))
+            || (config.hide_on_kbd && (cookie->evtype == XI_RawKeyPress))
+        ) {
```

Comment:
This approach definitely looks cleaner but I suspect it's also a little less efficient than simply using two separate if conditions. I'm just going to make the changes you suggested for now, since this level of optimisation probably isn't worth worrying about and it's ultimately a matter of design priorities anyway. 🙂